src/utils/sformat.fs

```diff
@@ -865,37 +865,74 @@ namespace Microsoft.FSharp.Text.StructuredFormat
-                                                      // A particular rule is that if the alternative property
-                                                      // returns a string, we turn off auto-quoting and esaping of
-                                                      // the string, i.e. just treat the string as display text.
-                                                      // This allows simple implementations of 
-                                                      // such as
-                                                      //
-                                                      //    [<StructuredFormatDisplay("{StructuredDisplayString}I")>]
-                                                      //    type BigInt(signInt:int, v : BigNat) =
-                                                      //        member x.StructuredDisplayString = x.ToString()
-                                                      //
-                                                      | :? string as s -> sepL s
-                                                      | _ -> sameObjL (depthLim-1) Precedence.BracketIfTuple alternativeObj
-                                                countNodes 0 // 0 means we do not count the preText and postText 
-                                                Some (leftL preText ^^ alternativeObjL ^^ rightL postText)
-                                            with _ -> 
-                                              None
-
+                                  let messageRegexPattern = @"^(.*?)(?<!\\){(.*?)(?<!\\)}(.*)$"
+                                  let rec buildObjMessageL (txt:string) (layouts:Layout list) =
+                                    
+                                    let replaceEscapedBrackets (txt:string) =
+                                      txt.Replace("\{", "{").Replace("\}", "}")
+
+                                    // to simplify support for escaped brackets, switch to using a Regex to simply parse the text as the following regex groups:
+                                    //  1) Everything up to the first opening bracket not preceded by a "\", lazily
+                                    //  2) Everything between that opening bracket and a closing bracket not preceded by a "\", lazily
+                                    //  3) Everything after that closing bracket
+                                    let m = System.Text.RegularExpressions.Regex.Match(txt, messageRegexPattern)
+                                    match m.Success with
+                                      | false when layouts.Length > 1 -> Some (spaceListL (List.rev layouts))
+                                      | false -> None
+                                      | true ->
+                                        let preText = replaceEscapedBrackets(m.Groups.get_Item(1).Value) // everything before the first opening bracket
```

Comment:
My noobishness with F# shines through here; I was naively trying to do `blah[n]` and ran into trouble, and recalled that `get_Item()` would work, but forgot that I was just missing that `.` to get the syntax I was looking for. Thanks!